### PR TITLE
feat: add wandb log top X models

### LINKED ISSUE #192: Make WandB log to disk then upload in batch
Only upload the top N models

Approach:
1. Get best runs
A. Save AUC + run name to a file in the evaluation script (requires being able to to extract the run name wandb needs)
B. Extract AUC from wandb offline-run (pretty hacky but works)
2. Run script that syncs top-n runs and moves the rest of the runs to an archive folder.

### PR DESCRIPTION
**Pull request by HLasse**
Fixes #192
Creates a logging file `.aucs/aucs.txt` that is populated during evaluation with run id and performance. This file is used by `src/sync_best_runs_to_wandb.py` to find the best performing runs, which are uploaded to wandb. All other runs are moved to `wandb/archive` and the logging file is deleted so users can safely start a new hyperparemeter search. 

- [X] I have battle-tested on Overtaci (RMAPPS1279)
- [ ] I have bumped the version as described in the [wiki](https://www.notion.so/Setting-up-Poetry-on-your-local-machine-76647d75e8fa4d9ba553cbc2a49946d9#dca1e0f2ce934a49acd1851aaf882d75)

## Notes for reviewers
Reviewers can skip X, but should pay attention to Y.

### CODE REVIEW

**Comment by MartinBernstorff** [comment]
Seems like the dependency tree is failing through `interpret` > `interpret-core` > `numba` > `llvmlite`. Trying to remove interpret to see if tests pass.

**Comment by MartinBernstorff** [comment]
Okay, new progress. The above tree did indeed cause problems, but new problems:

Pandoc doesn't conform to a PEP standard which is now enforced by pip: https://github.com/boisgera/pandoc/pull/49

I've temporarily disabled pandoc imports until we need them.

File: src/psycopt2d/evaluate_model.py [hunk 1]
```diff
@@ -74,6 +74,17 @@ def evaluate_model(
...
     run.log({"roc_auc_unweighted": auc})
 
+    # log AUC and run ID to a file to find the best run later
+    # Only create the file if it doesn't exists (will be auto-deleted/moved after
+    # syncing). This is to avoid creating a new file every time the script is run
+    # e.g. during a hyperparameter seacrch.
+    if cfg.project.wandb_mode == "offline":
+        if not AUC_LOGGING_FILE_PATH.exists():
+            AUC_LOGGING_FILE_PATH.parent.mkdir(parents=True, exist_ok=True)
+            AUC_LOGGING_FILE_PATH.touch()
```
Comment (MartinBernstorff) [inline src/psycopt2d/evaluate_model.py:84]: Maybe add the `run_id` and `auc` col headers here? :-)

File: src/psycopt2d/sync_best_runs_to_wandb.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        return list((Path() / "wandb").glob(f"*{run}*"))[0]
+
+    run_paths = [get_run_path(run) for run in runs]
+    for run in run_paths:
+        sync_single_run_to_wandb(run, project)
+
+
+def archive_wandb_runs() -> None:
+    """Move all wandb runs to an archive folder."""
+    wandb_path = Path() / "wandb"
```
Comment (MartinBernstorff) [inline src/psycopt2d/sync_best_runs_to_wandb.py:59]: Unsafe use of `Path()` – change to relative to `__file__`

File: src/psycopt2d/sync_best_runs_to_wandb.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    archive_path = wandb_path / "archive"
```
Comment (MartinBernstorff) [inline src/psycopt2d/sync_best_runs_to_wandb.py:60]: ❤️

File: src/psycopt2d/sync_best_runs_to_wandb.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    archive_path.mkdir(exist_ok=True)
+    for run in wandb_path.glob("*"):
+        if run == archive_path:
+            continue
+        run.rename(archive_path / run.name)
```
Comment (MartinBernstorff) [inline src/psycopt2d/sync_best_runs_to_wandb.py:65]: You've tested that this works? If so, awesome!

**Review by MartinBernstorff — APPROVED** [review]
If we add a top-level script here, it looks awesome!

**Comment by HLasse** [comment]
Waiting for the other PR to be merged to battle test on Overtaci

**Comment by MartinBernstorff** [comment]
PR has been merged and I've resolved conflicts with this branch. Should be ready for battle testing @HLasse.

**Comment by HLasse** [comment]
Alright, all functionality in this PR works perfectly fine on Overtaci. Only issue is related to the Altair saving functionality

**Pull request merged**